File: src/components/HousesCard.js
// components/HousesCard.js
import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet, Alert } from 'react-native';
import firestore from '@react-native-firebase/firestore';
import Ionicons from 'react-native-vector-icons/Ionicons';

import { useHouses } from '../contexts/HousesContext';

const HousesCard = ({ userId, houses }) => {
  const { currentHouseId, setCurrentHouseId } = useHouses();

  const handleLeave = id => {
    Alert.alert(
      'Leave house?',
      'Are you sure you want to leave this house?',
      [
        { text: 'Cancel', style: 'cancel' },
        {
          text: 'Leave',
          style: 'destructive',
          onPress: () => removeFromHouse(id),
        },
      ]
    );
  };

  const removeFromHouse = async houseId => {
    try {
      await firestore()
        .collection('houses')
        .doc(houseId)
        .update({
          members: firestore.FieldValue.arrayRemove(userId),
        });
      // If they just left the active house, clear selection
      if (houseId === currentHouseId) {
        setCurrentHouseId(null);
      }
    } catch {
      Alert.alert('Error', 'Could not leave house');
    }
  };

  const handleSelect = id => {
    setCurrentHouseId(id);
  };

  return (
    <View style={styles.card}>
      <Text style={styles.title}>Houses</Text>

      {houses.length === 0 ? (
        <Text style={styles.noHousesText}>You aren’t in any houses.</Text>
      ) : (
        houses.map(h => {
          const isActive = h.id === currentHouseId;
          return (
            <View
              key={h.id}
              style={[styles.houseItem, isActive && styles.activeHouseItem]}
            >
              <TouchableOpacity
                style={styles.houseNameContainer}
                onPress={() => handleSelect(h.id)}
              >
                <Text
                  style={[styles.houseName, isActive && styles.selectedText]}
                >
                  {h.houseName}
                </Text>
                {isActive && (
                  <Ionicons
                    name="checkmark"
                    size={18}
                    style={styles.checkIcon}
                  />
                )}
              </TouchableOpacity>

              <TouchableOpacity
                style={styles.leaveButton}
                onPress={() => handleLeave(h.id)}
              >
                <Text style={styles.leaveButtonText}>Leave</Text>
              </TouchableOpacity>
            </View>
          );
        })
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: '#1a1a1a',
    borderRadius: 16,
    padding: 20,
  },
  title: {
    fontSize: 24,
    color: '#fff',
    marginBottom: 20,
    textAlign: 'center',
    fontFamily: 'Montserrat-Bold',
  },
  noHousesText: {
    color: '#fff',
    textAlign: 'center',
    fontFamily: 'Montserrat-Regular',
  },
  houseItem: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 12,
    padding: 8,
  },
  activeHouseItem: {
    backgroundColor: '#2a2a2a',
    borderRadius: 8,
  },
  houseNameContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  houseName: {
    color: '#fff',
    fontSize: 16,
    fontFamily: 'Montserrat-Regular',
  },
  selectedText: {
    color: '#ae00ff',
    fontFamily: 'Montserrat-Bold',
  },
  checkIcon: {
    marginLeft: 6,
    color: '#ae00ff',
  },
  leaveButton: {
    backgroundColor: '#ff4d4d',
    borderRadius: 12,
    paddingVertical: 8,
    paddingHorizontal: 16,
    marginLeft: 'auto',
  },
  leaveButtonText: {
    color: '#fff',
    fontFamily: 'Montserrat-Medium',
  },
});

export default HousesCard;

File: flutter_app/lib/features/settings/widgets/houses_card.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:flutter_app/core/models/house.dart';
import 'package:flutter_app/core/providers/houses_provider.dart';

class HousesCard extends ConsumerWidget {
  final String userId;
  final List<House> houses;

  const HousesCard({super.key, required this.userId, required this.houses});

  Future<void> _leaveHouse(
    BuildContext context,
    String houseId,
    WidgetRef ref,
  ) async {
    final confirmed = await showDialog<bool>(
      context: context,
      builder: (context) => AlertDialog(
        title: const Text('Leave house?'),
        content: const Text('Are you sure you want to leave this house?'),
        actions: [
          TextButton(
            onPressed: () => Navigator.of(context).pop(false),
            child: const Text('Cancel'),
          ),
          TextButton(
            onPressed: () => Navigator.of(context).pop(true),
            style: TextButton.styleFrom(foregroundColor: Colors.red),
            child: const Text('Leave'),
          ),
        ],
      ),
    );

    if (confirmed == true) {
      try {
        await FirebaseFirestore.instance
            .collection('houses')
            .doc(houseId)
            .update({
              'members': FieldValue.arrayRemove([userId]),
            });

        // If current house is the one left, clear selection
        final currentHouseId = ref.read(currentHouseIdProvider);
        if (currentHouseId == houseId) {
          ref.read(currentHouseIdProvider.notifier).setHouseId(null);
        }
      } catch (e) {
        if (context.mounted) {
          ScaffoldMessenger.of(
            context,
          ).showSnackBar(SnackBar(content: Text('Error leaving house: $e')));
        }
      }
    }
  }

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final currentHouseId = ref.watch(currentHouseIdProvider);

    return Card(
      margin: const EdgeInsets.all(16),
      child: Padding(
        padding: const EdgeInsets.all(16),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            const Text(
              'Houses',
              style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
              textAlign: TextAlign.center,
            ),
            const SizedBox(height: 16),
            if (houses.isEmpty)
              const Text(
                'You aren’t in any houses.',
                textAlign: TextAlign.center,
              )
            else
              ...houses.map((house) {
                final isActive = house.id == currentHouseId;
                return Container(
                  margin: const EdgeInsets.only(bottom: 8),
                  decoration: isActive
                      ? BoxDecoration(
                          color: Theme.of(
                            context,
                          ).colorScheme.secondaryContainer,
                          borderRadius: BorderRadius.circular(8),
                        )
                      : null,
                  child: ListTile(
                    title: Text(
                      house.houseName,
                      style: TextStyle(
                        fontWeight: isActive
                            ? FontWeight.bold
                            : FontWeight.normal,
                        color: isActive
                            ? Theme.of(context).colorScheme.onSecondaryContainer
                            : null,
                      ),
                    ),
                    trailing: Row(
                      mainAxisSize: MainAxisSize.min,
                      children: [
                        if (isActive)
                          Icon(
                            Icons.check,
                            color: Theme.of(context).colorScheme.primary,
                          ),
                        IconButton(
                          icon: const Icon(
                            Icons.exit_to_app,
                            color: Colors.red,
                          ),
                          onPressed: () => _leaveHouse(context, house.id, ref),
                        ),
                      ],
                    ),
                    onTap: () {
                      ref
                          .read(currentHouseIdProvider.notifier)
                          .setHouseId(house.id);
                    },
                  ),
                );
              }),
          ],
        ),
      ),
    );
  }
}
